src/SDKs/Batch/DataPlane/Azure.Batch/GeneratedProtocol/PoolOperations.cs

```diff
@@ -621,6 +621,10 @@ internal PoolOperations(BatchServiceClient client)
             {
                 throw new ValidationException(ValidationRules.CannotBeNull, "this.Client.ApiVersion");
             }
+            if (pool == null)
+            {
+                pool = new PoolAddParameter();
```

Comment:
@matthchr similar comment about this not being reached because of the exception thrown above